libvast/vast/arrow_table_slice_builder.hpp

```diff
@@ -108,6 +109,10 @@ class arrow_table_slice_builder final : public table_slice_builder {
   /// Schema of the Record Batch corresponding to the layout.
   record_type flat_layout_;
 
+  /// The serialized layout can be cached because every builder instance only
+  /// produces slices of a single layout.
+  mutable std::vector<char> serialized_layout_cache_;
```

Comment:
```suggestion
  mutable std::vector<std::byte> serialized_layout_cache_;
```

YMMV: more idiomatic—but at the cost of a few casts.

Reply:
`char` is the type used by FlatBuffers internally, so we should stay with it here imo.

Reply:
@dominiklohmann is right, it has to be `std::vector<char>`, otherwise we need to introduce a bunch of needless casting.